packages/ra-core/src/controller/input/useReferenceInputController.ts

```diff
@@ -142,23 +136,19 @@ export const useReferenceInputController = (
     });
 
     // add current value to possible sources
-    let finalIds: Identifier[],
-        finalData: RecordMap<Record>,
-        finalTotal: number;
-    if (!referenceRecord || possibleValuesIds.includes(input.value)) {
-        finalIds = possibleValuesIds;
+    let finalData: Record[], finalTotal: number;
+    if (!referenceRecord || possibleValuesData.includes(input.value)) {
```

Comment:
Looks like a bug to me: `input.value` is of primitive type and `possibleValuesData` is an array of objects. Won't ever match.

This solves it in a pinch:

```diff
-    if (!referenceRecord || possibleValuesData.includes(input.value)) {
+    if (!referenceRecord || possibleValuesData.find(({ id }) => id === input.value) {
```

But I lack context to confidently suggest it as a fix. So just bringing it to your attention @fzaninotto. Let me know if it's as simple as that and I can put together a pull request.

If this doesn't quite make sense I can open a proper issue, but since it's `next` and actively being worked on I thought just dropping a comment here might be nicer.

Reply:
You're right, this is a bug! And the fix you suggest is correct. Would you mind opening a PR?

Commenting here is fine, it gives the proper context. The best would be to comment before the PR is merged ;)

Reply:
Thanks for quick response. #6987